launcher/ui/MainWindow.cpp

```diff
@@ -208,6 +208,13 @@ MainWindow::MainWindow(QWidget* parent) : QMainWindow(parent), ui(new Ui::MainWi
         exportInstanceMenu->addAction(ui->actionExportInstanceMrPack);
         exportInstanceMenu->addAction(ui->actionExportInstanceFlamePack);
         ui->actionExportInstance->setMenu(exportInstanceMenu);
+
+        auto shortcutInstanceMenu = new QMenu(this);
+        shortcutInstanceMenu->addAction(ui->actionCreateInstanceShortcutDesktop);
+        shortcutInstanceMenu->addAction(ui->actionCreateInstanceShortcutApplications);
+        shortcutInstanceMenu->addAction(ui->actionCreateInstanceShortcutOther);
```

Comment:
It would be nice to not create this menu if is on flatpack or the only available action is create shortcut other

Reply:
I've disabled them [here](https://github.com/PrismLauncher/PrismLauncher/pull/2966/files/b182a888aa3559ab3c0e8034533cc63040f84c34#diff-e2da2f07848e7be2b6912355a5d7f74ed7f971d0532cefdb55749055983e93dcR245-R251). Should I move it up?

Reply:
What I meant is to make this initialization as follows:
```
QList<QAction> actions={ui->actionCreateInstanceShortcutOther}
if !DesktopServices::isFlatpak(){
  QString applicationsDir = FS::getApplicationsDir();
    if (!applicationsDir.isEmpty()) actions.push_front(ui->actionCreateInstanceShortcutApplications)
 QString desktopDir = FS::getDesktopDir();
        if (desktopDir.isEmpty()) actions.push_front(ui->actionCreateInstanceShortcutDesktop)
}
if actions.lenght>1{ // we have at least two actions
   auto shortcutInstanceMenu = new QMenu(this);
for (auto a: actions)    shortcutInstanceMenu->addAction(a)
 ui->actionCreateInstanceShortcut->setMenu(shortcutInstanceMenu);
}
```
That way you do not need to double check the paths when you execute the actions and also have it be more minimal(do not display options that do not work)

PS. the code is more like a pseudocode so please adapt it (and you can even make it cleaner)

Reply:
Ooh I get it alright, will implement later

Reply:
Alright pushed it